contracts/LPFeeManager.sol

```diff
@@ -0,0 +1,161 @@
+            _swapTokens(pair.token0(), _outputToken, _to);
+            _swapTokens(pair.token1(), _outputToken, _to);
+        }
+    }
+
+    /// @notice Swapping of tokens to _outputToken
+    /// @param _token token to swap to _outputToken
+    /// @param _outputToken ERC20 token the LP tokens needs to be swapped to
+    /// @param _to address the tokens need to be transferred to.
+    function _swapTokens(
+        address _token,
+        address _outputToken,
+        address _to
+    ) private {
+        if (_token == _outputToken) {
+            IERC20 token = IERC20(_token);
+            uint256 balance = token.balanceOf(address(this));
+            token.transfer(_to, balance);
+            return;
+        }
+
+        if (factory.getPair(_token, _outputToken) == address(0)) {
+            address[] memory path = new address[](3);
+            for (uint256 i = 0; i < possiblePaths.length; i++) {
+                address pathToken = possiblePaths[i];
+                if (
+                    _token != pathToken &&
+                    _outputToken != pathToken &&
+                    factory.getPair(_token, pathToken) != address(0) &&
+                    factory.getPair(_outputToken, pathToken) != address(0)
+                ) {
+                    path[0] = _token;
+                    path[1] = pathToken;
+                    path[2] = _outputToken;
+                    break;
+                }
+            }
+            require(path[0] == _token, 'No path found');
+
+            IERC20 token = IERC20(_token);
+            uint256 amountIn = token.balanceOf(address(this));
+            token.approve(address(router), amountIn);
+            _safeSwap(amountIn, path, _to);
+        } else {
+            address[] memory path = new address[](2);
+            path[0] = _token;
+            path[1] = _outputToken;
+            IERC20 token = IERC20(_token);
+            uint256 amountIn = token.balanceOf(address(this));
+            token.approve(address(router), amountIn);
+            _safeSwap(amountIn, path, _to);
+        }
+    }
+
+    /// @notice Actual swapping of tokens to _outputToken
+    /// @param _amountIn amount of tokens to swap
+    /// @param _path path to follow for swapping
+    /// @param _to address the tokens need to be transferred to.
+    function _safeSwap(
+        uint256 _amountIn,
+        address[] memory _path,
+        address _to
+    ) internal virtual {
+        uint256[] memory amounts = router.getAmountsOut(_amountIn, _path);
+        uint256 amountOut = amounts[amounts.length - 1];
+
+        router.swapExactTokensForTokens(
+            _amountIn,
+            (amountOut * slippageFactor) / 1000,
+            _path,
+            _to,
+            block.timestamp + 600
+        );
+    }
+
+    /// @notice Change possible paths
+    /// @param _possiblePaths new list of possible paths
+    function changePossiblePaths(address[] memory _possiblePaths) public onlyOwner {
```

Comment:
An additional `function addPossiblePath(address _possiblePath)` would be nice so we can easily add a path without needing to compile all previous paths.